contracts/script/README.md

```diff
@@ -0,0 +1,142 @@
+# Deploying Upgradeable Smart Contracts
+
+Upgradeable Smart contracts are deployed with Openzeppelin Defender to enable a deployment strategy that is more secure
+and also uses a multi-sig Safe wallet. When deploying using openzeppelin the `defender` profile in the `foundry.toml`
+file is used.
+
+Note: One must set the environment variable `FOUNDRY_PROFILE` accordingly to use the appropriate profile in the
+`foundry.toml` settings. E.g.
+
+```bash
+export FOUNDRY_PROFILE=defender
+```
+
+The profile, `profile.default` is used by default. Setting the `FOUNDRY_PROFILE` variable overrides the `foundry.toml`
+settings.
+
+## Prerequisites
+
+1. Create a multisig wallet using [Safe](https://app.safe.global/welcome/accounts) on the network you'd like to deploy
+   to.
+2. In OpenZeppelin Defender, create an Approval Process that requires the multisig wallet you created above.
+   `Manage > Approval Processes`
+3. In OpenZeppelin Defender, create a deployment environment and use the approval process created in Step 2. Be sure to
+   copy the Defender secret and key, that is shown at the end of this step, into the .env file.
+
+## Deployments
+
+### Deploying the Fee Contract
+
+Steps:
+
+1. Run the Deployment Command This command requires you to go to OpenZeppelin Defender's UI to see the transaction.
```

Comment:
Where in the UI should I see this transaction?

I got this as output from `forge script`:

```
[⠊] Compiling...
[⠆] Compiling 92 files with 0.8.23
[⠒] Solc 0.8.23 finished in 17.23s
Compiler run successful with warnings:
Warning (2018): Function state mutability can be restricted to view
   --> contracts/lib/openzeppelin-foundry-upgrades/src/internal/Utils.sol:133:5:
    |
133 |     function getOutDir() internal returns (string memory) {
    |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to pure
  --> contracts/test/PlonkVerifier.t.sol:29:5:
   |
29 |     function sanitizeScalarField(uint256 a) public view returns (uint256) {
   |     ^ (Relevant source part starts here and spans across multiple lines).

2024-03-18T14:16:10.827858Z  WARN foundry_evm_core::fork::cache: Failed to read cache file err=Os { code: 2, kind: NotFound, message: "No such file or directory" } path="/home/lulu/.foundry/cache/rpc/sepolia/5511676"
```

Reply:
Yes so at this point, you'd go into openzepelin defender (OPD) and you'd see the transaction waiting to be signed by the wallets in the multisig 'safe'. I just checked OPD to see if there is a transaction waiting but I don't. I'll DM you

Reply:
We noticed that my initial attempt wasn't properly indexed but it worked when I tried again a few hours later.